Assets/LeapMotionModules/Interaction/InteractionBehaviour.cs

```diff
@@ -0,0 +1,711 @@
+      FixedUpdateHovering();
+      FixedUpdatePrimaryHovering();
+      FixedUpdateContact();
+      FixedUpdateGrasping();
+
+      FixedUpdateCollisionMode();
+    }
+
+    // TODO: Currently this gets the distance from the point to this transform, but this should
+    // probably incorporate distance to the rigidbody.
+    protected virtual float GetInteractionDistanceToPoint(Vector3 point) {
+      return Vector3.Distance(point, this.transform.position);
+    }
+
+    #region Hovering
+
+    // Logistics for providing per-object (instead of per-hand) Hover callbacks.
+    private Hand  _closestHoveringHand = null;
+    private float _closestHoveringHandDistance = float.PositiveInfinity;
+    private Hand  _closestJustStoppedHoveringHand = null; // Provided for OnObjectHoverEnd.
+    private int   _hoveringHandsCountLastFrame = 0;
+    private int   _hoveringHandsCount = 0;
+
+    private Hand _publicClosestHoveringHand = null;
+
+    // Runs after InteractionHands have done FixedUpdateHand.
+    private void FixedUpdateHovering() {
+      if (_hoveringHandsCount > 0) {
+        if (_hoveringHandsCountLastFrame == 0) {
+          _publicClosestHoveringHand = _closestHoveringHand;
+          OnObjectHoverBegin(_closestHoveringHand);
+        }
+        else {
+          _publicClosestHoveringHand = _closestHoveringHand;
+          OnObjectHoverStay(_closestHoveringHand);
+        }
+      }
+      else if (_hoveringHandsCountLastFrame > 0) {
+        _closestHoveringHand = null;
+        OnObjectHoverEnd(_closestJustStoppedHoveringHand);
+        _publicClosestHoveringHand = null;
+      }
+
+      _hoveringHandsCountLastFrame = _hoveringHandsCount;
+      _closestHoveringHand = null;
+      _closestHoveringHandDistance = float.PositiveInfinity;
+    }
+
+    public override float GetDistance(Vector3 worldPosition) {
+      // TODO: Should probably get distance from the InteractionBehaviour's colliders. Probably has to wait until 5.6 (Physics.ClosestPoint)
+      return GetInteractionDistanceToPoint(worldPosition);
+    }
+
+    public override void HoverBegin(Hand hand) {
+      _hoveringHandsCount++;
+      EvaluateHoverCloseness(hand);
+
+      OnHoverBegin(hand);
+    }
+
+    public override void HoverStay(Hand hand) {
+      EvaluateHoverCloseness(hand);
+
+      OnHoverStay(hand);
+    }
+
+    private void EvaluateHoverCloseness(Hand hand) {
+      float handDistance = GetInteractionDistanceToPoint(hand.PalmPosition.ToVector3());
+
+      if (_closestHoveringHand == null) {
+        _closestHoveringHand = hand;
+        _closestHoveringHandDistance = handDistance;
+      }
+      else {
+        if (handDistance < _closestHoveringHandDistance) {
+          _closestHoveringHand = hand;
+          _closestHoveringHandDistance = handDistance;
+        }
+      }
+    }
+
+    public override void HoverEnd(Hand hand) {
+      _hoveringHandsCount--;
+      if (_hoveringHandsCount == 0) {
+        _closestJustStoppedHoveringHand = hand;
```

Comment:
Same question as before, this is named closest but what makes it the closest?  There are no more hovering hands, so isn't this just the most recent hovering hand?